File: app/src/main/java/com/example/hcwong/testproject/Main/MainActivity.java
package com.example.hcwong.testproject.Main;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.hcwong.testproject.Library.DepthPageTransformer;
import com.example.hcwong.testproject.Main.ListNews.ListFragment;
import com.example.hcwong.testproject.Main.CreateNews.LocalFragment;
import com.example.hcwong.testproject.Model.Article;
import com.example.hcwong.testproject.Main.News.NewsFragment;
import com.example.hcwong.testproject.R;

public class MainActivity extends AppCompatActivity implements LocalFragment.OnListFragmentInteractionListener, NewsFragment.OnListFragmentInteractionListener, ListFragment.OnListFragmentInteractionListener{

    private static final int NUM_PAGES=3;
    private ViewPager pager;
    private PagerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        pager=findViewById(R.id.pager);
        adapter= new PagerAdapter(getSupportFragmentManager());
        pager.setAdapter(adapter);
        pager.setPageTransformer(true,new DepthPageTransformer());

        TabLayout tabLayout = findViewById(R.id.sliding_tabs);
        tabLayout.setupWithViewPager(pager);

    }


    @Override
    public void onBackPressed() {
        if(pager.getCurrentItem()==0)
            super.onBackPressed();
        else
            pager.setCurrentItem(pager.getCurrentItem()-1);
    }

    @Override
    public void onListFragmentInteraction(Article item) {

    }

    private class PagerAdapter extends FragmentPagerAdapter {
        public PagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            if (position == 0)
                return NewsFragment.newInstance();
            else if (position == 1)
                return  LocalFragment.newInstance();
            else if (position == 2)
                return  ListFragment.newInstance();
            else
                return  LocalFragment.newInstance();
        }

        @Override
        public int getCount() {
            return NUM_PAGES;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            if (position == 0)
                return getResources().getString(R.string.news);
            else if (position == 1)
                return getResources().getString(R.string.add_news);
            else if (position == 2)
                return getResources().getString(R.string.list_news);
            else
                return getResources().getString(R.string.non);
        }
    }

}

File: app/src/main/java/com/example/hcwong/testproject/Main/MainActivity.kt
package com.example.hcwong.testproject.Main

import android.support.design.widget.TabLayout
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

import com.example.hcwong.testproject.Library.DepthPageTransformer
import com.example.hcwong.testproject.Main.ListNews.ListFragment
import com.example.hcwong.testproject.Main.CreateNews.LocalFragment
import com.example.hcwong.testproject.Model.Article
import com.example.hcwong.testproject.Main.News.NewsFragment
import com.example.hcwong.testproject.R

class MainActivity : AppCompatActivity(), LocalFragment.OnListFragmentInteractionListener, NewsFragment.OnListFragmentInteractionListener, ListFragment.OnListFragmentInteractionListener {
    private var pager: ViewPager? = null
    private var adapter: PagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        pager = findViewById(R.id.pager)
        adapter = PagerAdapter(supportFragmentManager)
        pager!!.adapter = adapter
        pager!!.setPageTransformer(true, DepthPageTransformer())

        val tabLayout = findViewById<TabLayout>(R.id.sliding_tabs)
        tabLayout.setupWithViewPager(pager)

    }


    override fun onBackPressed() {
        if (pager!!.currentItem == 0)
            super.onBackPressed()
        else
            pager!!.currentItem = pager!!.currentItem - 1
    }

    override fun onListFragmentInteraction(item: Article) {

    }

    private inner class PagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
        override fun getItem(position: Int): Fragment {
            return if (position == 0)
                NewsFragment.newInstance()
            else if (position == 1)
                LocalFragment.newInstance()
            else if (position == 2)
                ListFragment.newInstance()
            else
                LocalFragment.newInstance()
        }

        override fun getCount(): Int {
            return NUM_PAGES
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return if (position == 0)
                resources.getString(R.string.news)
            else if (position == 1)
                resources.getString(R.string.add_news)
            else if (position == 2)
                resources.getString(R.string.list_news)
            else
                resources.getString(R.string.non)
        }
    }

    companion object {

        private val NUM_PAGES = 3
    }

}
